src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/Vector256_1.cs

```diff
@@ -267,10 +268,26 @@ internal string DisplayString
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static Vector256<T> operator *(Vector256<T> left, Vector256<T> right)
         {
-            return Vector256.Create(
-                left._lower * right._lower,
-                left._upper * right._upper
-            );
+            if ((typeof(T) == typeof(ulong) || typeof(T) == typeof(long)) && Avx2.IsSupported)
```

Comment:
Nice!
Perhaps, worth leaving a comment that this path is not taken when avx-512 is presented?